include/aspect/simulator_access.h

```diff
@@ -51,14 +51,18 @@ namespace aspect
 {
   using namespace dealii;
 
-  // forward declaration
+  // forward declarations:
   template <int dim> class Simulator;
   template <int dim> struct SimulatorSignals;
   template <int dim> class LateralAveraging;
   namespace HeatingModel
   {
     template <int dim> class Manager;
   }
+  namespace AdiabaticConditions
+  {
+    template <int dim> class Interface;
```

Comment:
interface.h now needs simulator_access.h (because it is derived from it), which means we need to be able to parse simulator_access.h without knowing interface.h. If you scroll down, we need to know AdiabaticConditions. Circular dependencies, yay.